src/Application/HttpModels/SendRequest.php

```diff
@@ -59,4 +60,9 @@ public function __construct()
         $this->id = (Uuid::uuid4())->toString();
         $this->env = getenv('APP_ENV');
     }
+
+    public function getMessageGroupId(): ?string
```

Comment:
very minor but may as well, unless you think its useful to keep the return nullable.

```suggestion
    #[Override]
    public function getMessageGroupId(): string
```

Reply:
Nope, makes sense! Forgot we had this attribute now. I think it need importing or `\` so added with the latter in https://github.com/thebiggive/mailer/pull/145/commits/9b012f2ead8d06044270fd2152617ce9660bbab7